tests/kola/rpm-ostree/replace-rt-kernel

```diff
@@ -19,64 +22,54 @@ rhelver=$(. /usr/lib/os-release && echo "${RHEL_VERSION}")
 major=$(echo "${rhelver}" | cut -f 1 -d .)
 baseurl=
 case "${major}" in
-    8)
-        # TODO: why the heck does centos only support insecure http?!?
-        baseurl=http://mirror.centos.org/centos/8-stream/
-        # TODO avoid hardcoding versions
-        target_kver=4.18.0-383.el8
-        target_kver_rt=4.18.0-394.rt7.179.el8
-        ;;
     9)
         # TODO: why the heck does centos only support insecure http?!?
         baseurl=http://mirror.stream.centos.org/9-stream/
-        # TODO avoid hardcoding versions
-        target_kver=5.14.0-196.el9
-        target_kver_rt=5.14.0-201.rt14.202.el9
+        repo_url=https://raw.githubusercontent.com/openshift/os/bf4383e4929ad9abb67d508a7ffa7cf45e8de5ca/c9s.repo
         ;;
     *)  fatal "Unhandled RHEL_VERSION=${rhelver}"
         ;;
 esac
 
+# setup repos
+runv rm -rf /etc/yum.repos.d/*
+runv curl -v "${repo_url}" -o /etc/yum.repos.d/cs.repo
+runv curl -v https://centos.org/keys/RPM-GPG-KEY-CentOS-Official -o /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Official
+runv sed -i 's|^gpgkey.*|gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Official|' /etc/yum.repos.d/cs.repo
+
 case "${AUTOPKGTEST_REBOOT_MARK:-}" in
 "")
     echo "Testing overriding with previous CentOS Stream kernel"
-    kver=$(uname -r)
-    if test "${kver}" = "${target_kver}"; then
-        fatal "Already in ${target_kver}"
-    fi
-    tmpdir=$(mktemp -d -p /var/tmp)
-    pushd "${tmpdir}"
-    runv curl -sSLf --remote-name-all "${baseurl}/BaseOS/${basearch}/os/Packages/kernel"{,-core,-modules,-modules-extra}"-${target_kver}.${basearch}.rpm"
-    runv rpm-ostree override replace ./*.rpm
-    popd
-    rm "${tmpdir}" -rf
+    runv rm -rf /etc/yum.repos.d/*
+    runv curl -v "${repo_url}" -o /etc/yum.repos.d/cs.repo
+    runv curl -v https://centos.org/keys/RPM-GPG-KEY-CentOS-Official -o /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Official
```

Comment:
This isn't a big deal right now, but in the general case you always want to use `curl -L` to follow redirects (here and everywhere). 

And we also want `-f` to fail if there's a HTTP error (otherwise we'll output the error HTML successfully into what should be a GPG key).

We had these flags in the `curl -sSLf` that was removed below.

Reply:
fixed.